test/memory.test.js

```diff
@@ -695,6 +695,38 @@ describe('Memory connector', function() {
 
   });
 
+  describe('findOrCreate', function() {
+    var ds, Cars;
+    before(function() {
+      ds = new DataSource({connector: 'memory'});
+      Cars = ds.define('Cars', {
+        color: String
+      });
+    });
+
+    it('should create a specific object once and in the subsequent calls it should find it', function(done) {
+      var creationNum = 0;
+      async.times(100, function(n, next) {
+        var initialData = {color: 'white'};
+        var query = {'where': initialData};
+        Cars.findOrCreate(query, initialData, function(err, car, created) {
+          if (created) creationNum++;
+          next(err, car);
+        });
+      }, function(err, cars) {
+        if (err) done(err);
+        Cars.find(function(err, data) {
+          if (err) done(err);
+          data.length.should.equal(1);
+          data[0].color.should.equal('white');
+          creationNum.should.equal(1);
+          done();
+        });
+      });
+    });
+  });
+
+
```

Comment:
This test was failing before this PR and now it is passing; however, I'm not sure whether I defined this test in the right place or not,...